### PR TITLE
[gitlab-housekeeping] merge without waiting for pending jobs

### PR DESCRIPTION
**Pull request by maorfr**
part of https://issues.redhat.com/browse/APPSRE-2484

if a job is cancelled, in many cases the gitlab pipeline will hang and wait forever. we want to merge based on the latest result only, as each run already cancelled the previous one.

### CODE REVIEW

File: reconcile/gitlab_housekeeping.py [hunk 1]
```diff
@@ -165,16 +166,17 @@ def merge_merge_requests(dry_run, gl, merge_limit, rebase, insist=False):
...
-            # running, pending, success, failed, canceled, skipped
-            incomplete_pipelines = \
-                [p for p in pipelines
-                 if p['status'] in ['running', 'pending']]
-            if incomplete_pipelines:
-                if insist:
-                    raise Exception(f'insisting on {merge_label}')
-                else:
-                    continue
+            if wait_for_pipeline:
```
Comment (apahim) [inline reconcile/gitlab_housekeeping.py:169]: Shouldn't we wait for running?

Comment (maorfr) [reply reconcile/gitlab_housekeeping.py:169]: as long as the last one passed, we don't really care about previous ones...
plus, the disconnection may also mean that jobs that are "running" are not really running anymore.

Comment (apahim) [reply reconcile/gitlab_housekeeping.py:169]: got it

**Review by apahim — APPROVED** [review]
lgtm

**Pull request merged**